src/getting-started.md

```diff
@@ -0,0 +1,477 @@
+can do the following:
+
+```sh
+# first time
+git submodule update --init --recursive
+
+# subsequent times (to pull new commits)
+git submodule update
+```
+
+### Configuring the Compiler
+
+The compiler has a configuration file which contains a ton of settings. We will
+provide some recommendations here that should work for most, but [check out
+this chapter for more info][config].
+
+[config]: ./building/how-to-build-and-run.html#create-a-configtoml
+
+In the top level of the repo:
+
+```sh
+cp config.toml.example config.toml
+```
+
+Then, edit `config.toml`. You may want to search for, uncomment, and update
+the following settings:
+
+- `debug = true`: enables debug symbols and `debug!` logging, takes a bit longer to compile.
+- `incremental = true`: enables incremental compilation of the compiler itself,
+  which can significantly speed things up.  This is turned off by default
+  because it's technically unsound; sometimes it will cause weird crashes.
+  Also, it can consume a lot of disk space. This has the same effect as the
+  `-i` or `--incremental` flags.
+- `llvm-config`: enables building with system LLVM. [See this chapter][sysllvm]
+  for more info. This avoids building LLVM, which can take a while (45 minutes
+  on my laptop; others have reported 15 minutes or faster with incremental
+  compilation).
+
+[sysllvm]: ./building/suggested.html#building-with-system-llvm
+
+### `./x.py` Intro
+
+`rustc` is a _bootstrapping_ compiler, which means that it is written in Rust
+and thus needs to be compiled by itself. So where do you
+get the original compiler from? We use the current beta compiler
+to build a new compiler. Then, we use that compiler to build itself. Thus,
+`rustc` has a 2-stage build. You can read more about bootstrapping
+[here][boot], but you don't need to know much more to contribute.
+
+[boot]: ./building/bootstrapping.md
+
+We have a special tool `./x.py` that drives this process. It is used for
+building the compiler, the standard libraries, and `rustdoc`. It is also used
+for driving CI and building the final release artifacts.
+
+Unfortunately, a proper 2-stage build takes a long time depending on your
+hardware, but it is the only correct way to build everything (e.g. it's what
+the CI and release processes use). **However, in most cases, you can get by
+without a full 2-stage build**. In the following section, we give instructions
+for how to do "the correct thing", but then we also give various tips to speed
+things up.
+
+### Building and Testing `rustc`
+
+To do a full 2-stage build of the whole compiler, you should run this (after
+updating `config.toml` as mentioned above):
+
+```sh
+./x.py build
+```
+
+In the process, this will also necessarily build the standard libraries, and it
+will build `rustdoc` (which doesn't take too long).
+
+To build and test everything:
+
+```sh
+./x.py test
+```
+
+For most contributions, you only need to build stage 1, which saves a lot of time:
```

Comment:
I think it would also be nice to switch the default to `--stage 1` at some point and make `--stage 2` opt-in. I think the only regular use of stage 2 is for CI (??)

Reply:
So, I definitely agree that I think we should make the "default" commands the ones that most people need rather than what release does (e.g. `./x.py build` would do a stage-1 incremental build), but it also seems odd to have the default command be technically unsound...

I'm not really sure what to make of this situation...

Reply:
Oh, also, I forgot to mention `--bless`...

Reply:
I vote for the most common commands first, with a disclaimer that they may not work and you should ask on Zulip.

Since it’s a getting started guide, I think the command that beginners will use more should be at the top.

Maybe mention that after a major rust release, you should run the sound command since now the older part was compiled with a different compiler? At least, I think I needed to do that after 1.44

Reply:
Maybe say not to worry about using less often since you can just compare the git diff later? At first I thought it was better to edit stderr manually, until I realized I could just diff later

Reply:
I added Niko's table to the top of the section.